ICSharpCode.Decompiler/ILAst/ILAstBuilder.cs

```diff
@@ -539,7 +539,7 @@ void ConvertLocalVariables(List<ByteCode> body)
 				if (!optimize || varDef.IsPinned || uses.Any(b => b.VariablesBefore[varDef.Index].UnknownDefinition || (b.Code == ILCode.Ldloca && !IsDeterministicLdloca(b)))) {				
 					newVars = new List<VariableInfo>(1) { new VariableInfo() {
 						Variable = new ILVariable() {
-							Name = string.IsNullOrEmpty(varDef.Name) ? "var_" + varDef.Index : varDef.Name,
+							Name = varDef.ToString(),
```

Comment:
Looks like I used that extension method commented-on above, [here](https://github.com/airbreather/ILSpy/commit/4d18cd33e29de480f172b0b23aa4a1fefab845fa#diff-e0c25c650e7242249a3c2390e29cd889R542) too.